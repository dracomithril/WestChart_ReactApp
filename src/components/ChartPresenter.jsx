/**
 * Created by Gryzli on 06.04.2017.
 */
import React from "react";
import PropTypes from 'prop-types';
import {Tab, Tabs} from "react-bootstrap";
import Summary from "./Summary";
import ChartTable from "./ChartTable";
import ChartHeader from "./ChartHeader";
import SpotifySearch from "./SpotifySearch";

const {sorting, filterChart} = require('./../utils');

export default class ChartPresenter extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            show: false
        }
    }

    /*istanbul ignore next*/
    componentWillUnmount() {
        console.log('component ChartPresenter unmounted');
    }

    /*istanbul ignore next*/
    componentDidMount() {
        console.log('component ChartPresenter did mount');
    }

    render() {
        const {store} = this.context;
        const {list_sort, chart} = store.getState();
        const {view_chart, error_days, newsLetter} = chart.length > 0 ? filterChart(store) : {view_chart: [], error_days: [], newsLetter:[]};
        let selected = view_chart.filter((elem) => elem.selected);
        sorting[list_sort](selected);

        return (<div>
            <Tabs defaultActiveKey={0} id="chart_tabs">
                <Tab eventKey={0} title={<i className="fa fa-facebook">Posts</i>}>
                    <ChartHeader error_days={error_days} view_chart={view_chart}/>
                    <ChartTable data={view_chart}/>
                </Tab>
                <Tab eventKey={1} title={<i className="fa fa-spotify" id="chart_playlist_tab">Playlist</i>}>
                    <SpotifySearch selected={selected}/>
                </Tab>
                <Tab eventKey={2} title={<i className="fa fa-list">Summary</i>}>
                    <Summary selected={selected}/>
                </Tab>
                <Tab eventKey={3} title={<i className="fa fa-table">News Letters</i>}>
                    <ChartTable data={newsLetter}/>
                </Tab>
            </Tabs>
        </div>);
    }
}
ChartPresenter.contextTypes = {
    store: PropTypes.object
};
ChartPresenter.propTypes = {
};