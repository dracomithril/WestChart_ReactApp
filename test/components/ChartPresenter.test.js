/**
 * Created by XKTR67 on 4/19/2017.
 */
import React from 'react';
import PropTypes from 'prop-types';
import ChartPresenter from '../../src/front/components/ChartPresenter';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
const initial_state =require('../data/initial_state.json');
const mockStore = configureMockStore([]);
describe('<ChartPresenter/>', () => {
    it('renders without crashing ChartPresenter', () => {
        const store = mockStore(initial_state);
        const wrapper = shallow(
            <ChartPresenter view_chart={[]}/>, {
                context: {store},
                childContextTypes: {store: PropTypes.object}
            }
        );
        expect(shallowToJson(wrapper)).toMatchSnapshot();

    });

});
