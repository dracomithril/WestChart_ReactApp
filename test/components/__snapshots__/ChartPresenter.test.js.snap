// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartPresenter/> renders without crashing ChartPresenter 1`] = `
<div>
  <ChartHeader
    count={0}
    error_days={Array []}
  />
  <ChartTable
    data={Array []}
    error_days={Array []}
  />
  <Summary
    selected={Array []}
  />
  <SpotifySearch
    selected={Array []}
  />
</div>
`;
